From 92e1131fb00c5fd16911ea98777b872dbd0181b1 Mon Sep 17 00:00:00 2001
From: Pino Toscano <toscano.pino@tiscali.it>
Date: Mon, 20 Apr 2020 08:31:02 +0200
Subject: [PATCH] Use macOS API only on macOS

__MACH__ matches any Mach-based OS, which means both macOS and GNU/Hurd;
since the API used is really specific to macOS, then also check for
__APPLE__.
---
 libs/test/include/boost/test/utils/timer.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/test/include/boost/test/utils/timer.hpp b/libs/test/include/boost/test/utils/timer.hpp
index 1241692e..78dfa708 100644
--- a/libs/test/include/boost/test/utils/timer.hpp
+++ b/libs/test/include/boost/test/utils/timer.hpp
@@ -18,7 +18,7 @@
 
 # if defined(_WIN32) || defined(__CYGWIN__)
 #   define BOOST_TEST_TIMER_WINDOWS_API
-# elif defined(__MACH__)// && !defined(CLOCK_MONOTONIC)
+# elif defined(__MACH__) && defined(__APPLE__)// && !defined(CLOCK_MONOTONIC)
 #   // we compile for all macs the same, CLOCK_MONOTONIC introduced in 10.12
 #   define BOOST_TEST_TIMER_MACH_API
 # else
-- 
2.26.2

